src/main/java/rip/bolt/ingame/ranked/RequeueManager.java

```diff
@@ -0,0 +1,61 @@
+package rip.bolt.ingame.ranked;
+
+import static tc.oc.pgm.lib.net.kyori.adventure.text.Component.text;
+
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.ClickType;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import rip.bolt.ingame.Ingame;
+import rip.bolt.ingame.api.definitions.BoltResponse;
+import rip.bolt.ingame.utils.Messages;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.ObserverInteractEvent;
+import tc.oc.pgm.lib.net.kyori.adventure.text.format.NamedTextColor;
+
+public class RequeueManager implements Listener {
+
+  private static final ItemStack RED_DYE = createRequeueItem(1);
+  private static final ItemStack GREEN_DYE = createRequeueItem(2);
+
+  private static ItemStack createRequeueItem(int data) {
+    ItemStack item = new ItemStack(Material.INK_SACK, 1, (short) data);
+    ItemMeta meta = item.getItemMeta();
+    meta.setDisplayName(ChatColor.GOLD + "" + ChatColor.BOLD + "Click to requeue");
+    item.setItemMeta(meta);
+
+    return item;
+  }
+
+  public void sendRequeueMessage(MatchPlayer player) {
+    player.getInventory().setItem(2, RED_DYE);
+    player.sendMessage(Messages.requeue());
+  }
+
+  @EventHandler
+  public void onRequestRequeue(ObserverInteractEvent event) {
+    if (!(event.getClickType() == ClickType.LEFT || event.getClickType() == ClickType.RIGHT))
+      return;
+
+    if (event.getClickedItem().equals(RED_DYE)) requestRequeue(event.getPlayer());
+  }
+
+  public void requestRequeue(MatchPlayer player) {
```

Comment:
Because red dye does not instantly get replaced, however it does instantly call request requeue, it means players can spam it very easily, clicking it 10 times before it manages to succeed or fail, you'll need some cooldown on this